hudi-io/src/main/java/org/apache/hudi/storage/StorageSchemes.java

```diff
@@ -86,6 +88,11 @@ public enum StorageSchemes {
   // Hopsworks File System
   HOPSFS("hopsfs", false, true);
 
+  // list files may bring pressure to storage with centralized meta service like HDFS.
+  // when we want to get only part of files under a directory rather than all files, use getStatus may be more friendly than listStatus.
+  // here is a trade-off between rpc times and throughput of storage meta service
+  private static final Set<String> LIST_STATUS_FRIENDLY_SCHEMES = new HashSet<>(Arrays.asList(FILE.scheme, S3.scheme, S3A.scheme, GCS.scheme));
```

Comment:
Add it as part of the scheme definition
```
StorageSchemes(String scheme, Boolean isWriteTransactional, Boolean supportAtomicCreation, Boolean isListStatusFriendly)
```

Reply:
it looks like you said for the early commits, but I think the current change is better because only one invoker uses it, we can generalize it in the future if there are more use cases.

Reply:
OK, I see your point.  If this is not going to be widely used, I agree that we can use the static variables.